### PR TITLE
Address {issue}10811[10811] - Add configuration knob for auto-discover default.disable 

### PR DESCRIPTION
**Pull request by alakahakai**
Address #10811. Closed old PR #10818.

### CODE REVIEW

**Comment by alakahakai** [comment]
jenkins, test this

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -241,6 +241,7 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...master[Check the HEAD d
 - Populate more ECS fields in the Suricata module. {pull}10006[10006]
 - Add ISO8601 timestamp support in syslog metricset. {issue}8716[8716] {pull}10736[10736]
 - Add more info to message logged when a duplicated symlink file is found {pull}10845[10845]
+- Add configuration knob for auto-discover hints to control whether log harvesting is enabled for the pod/container. {issue}10811[10811]
```
Comment (kaiyan-sheng) [inline CHANGELOG.next.asciidoc]: {pull}10911[10911] will be good to add here.

Comment (alakahakai) [reply CHANGELOG.next.asciidoc]: Sure, will do before the final merge. This was added before the pull number was generated. Thanks!

Comment (alakahakai) [reply CHANGELOG.next.asciidoc]: Please approve if everything is OK. I'd like to get this in ASAP, thanks.

File: libbeat/autodiscover/providers/docker/config.go [hunk 2]
```diff
@@ -25,14 +25,15 @@ import (
...
+	Prefix         string                  `config:"prefix"`
+	HintsEnabled   bool                    `config:"hints.enabled"`
+	DefaultDisable bool                    `config:"default.disable"`
+	Builders       []*common.Config        `config:"builders"`
+	Appenders      []*common.Config        `config:"appenders"`
+	Templates      template.MapperSettings `config:"templates"`
+	Dedot          bool                    `config:"labels.dedot"`
 }
 
 func defaultConfig() *Config {
```
Comment (kaiyan-sheng) [inline libbeat/autodiscover/providers/docker/config.go]: Should default.disable be added in defaultConfig?

**Comment by kaiyan-sheng** [comment]
Tested this PR and it works. It would be good to add some documentation on `default.disable`. Also, should `default.disable` be added in the defaultConfig?

File: libbeat/autodiscover/providers/docker/config.go (hunk 2 above)
Comment (alakahakai) [reply libbeat/autodiscover/providers/docker/config.go]: The default value is false in GO, hence no need to add it here.

**Comment by alakahakai** [comment]
Will try to update the documentation. Thanks for the suggestion!

**Comment by alakahakai** [comment]
jenkins, test this

**Comment by exekias** [comment]
any plan to backport this? sounds like an interesting feature to have in 7.x

**Comment by Evesy** [comment]
Do you know which Filebeat release this PR is going to make it into?

I was interested to see it released in the 6.x series since I don't believe 7.x beats are compatible with 6.x ES clusters, can't find any release notes that mention this change

**Comment by exekias** [comment]
Hi @Evesy, this will land in 7.2, to be released soon. As far as I know, Filebeat 7.2 should be able to ship to 6.X ES. Please take into account that we updated this feature before the release, more info can be found here: https://github.com/elastic/beats/pull/12193

**Pull request merged**